easybuild/easyblocks/g/gromacs.py

```diff
@@ -102,6 +105,15 @@ def get_gromacs_arch(self):
             # According to [2] the performance difference between SSE2 and SSE4.1 is minor on x86
             # and SSE4.1 is not supported by AMD Magny-Cours[1].
             res = 'SSE2'
+        elif optarch == OPTARCH_GENERIC:
+            cpu_arch = get_cpu_architecture()
+            if cpu_arch == X86_64:
+                res = 'SSE2'
+            else:
+                res = 'None'
+        elif optarch:
+            print_warning("OPTARCH set to %s but not used."
+                          "Compiling GROMACS for the current host architecture", optarch)
```

Comment:
@victorusu You need to use `%` to format the string when using `print_warning` (while just passing the variables as argument does work with `self.log`...).

We can enhance `print_warning` to also handle this, but as it is now you'll run into a crash, so please change.

Also, I'd both print & log (since the printed message will otherwise not be included in the log...)

```python
warn_msg = "--optarch configuration setting set to %s but not taken into account; " % optarch
warn_msg += "compiling GROMACS for the current host architecture (i.e. the default behavior)"
self.log.warning(warn_msg)
print_warning(warn_msg)